### PR TITLE
Authenthicator blacklist

### PR DESCRIPTION
**Pull request by rkdarst**
I was looking for low-threshold useful stuff to do and improved #1739: added the test and addressed the comment.  I haven't tested in production, and probably won't in the near future unless someone requests it.

- Closes: #1739

### CODE REVIEW

File: jupyterhub/tests/test_auth.py [hunk 1]
```diff
@@ -103,6 +103,30 @@ def getgrnam(name):
...
+    assert authorized['name'] == 'wash'
+
+    authenticator = MockPAMAuthenticator(blacklist={'wash'})
+    authorized = yield authenticator.get_authenticated_user(None, {
+        'username': 'wash',
+        'password': 'wash',
+    })
+    assert authorized is  None
+
+    authenticator = MockPAMAuthenticator(blacklist={'wash'}, whitelist={'wash', 'kaylee'})
```
Comment (minrk) [inline jupyterhub/tests/test_auth.py]: Maybe remove 'wash' from whitelist. I don't want a user being in both whitelist and blacklist to be supported.

File: jupyterhub/auth.py [hunk 2]
```diff
@@ -88,6 +88,18 @@ class Authenticator(LoggingConfigurable):
...
+    blacklist = Set(
+        help="""
+        Blacklist of usernames that are not allowed to log in.
+
+        Use this with supported authenticators to restrict which users can not log in. This is an
+        additional blacklist that further restricts users, beyond whatever restrictions the
+        authenticator has in place.
+
+        If empty, does not perform any additional restriction.
+        """
```
Comment (minrk) [inline jupyterhub/auth.py]: Can you add 

    .. versionadded: 0.9

to the end of the docstring here and in the new method ?

File: jupyterhub/tests/test_auth.py (hunk 1 above)
Comment (rkdarst) [reply jupyterhub/tests/test_auth.py]: My thought was shouldn't be supported, but it must either default-deny (tested for) or raise an error (not tested)...

Comment (minrk) [reply jupyterhub/tests/test_auth.py]: OK

Comment (willingc) [reply jupyterhub/tests/test_auth.py]: We might consider (doesn't need to be in this PR) testing several combinations for expected behavior: user not in either list; user in one or the other; user in both

**Comment by minrk** [comment]
Great!

**Comment by willingc** [comment]
@rkdarst Thanks :cake:

**Pull request merged**